test/e2e/functional/tests/e2e.go

```diff
@@ -73,47 +74,57 @@ var _ = Describe("metallb", func() {
 
 		It("should have MetalLB pods in running state", func() {
 			By("checking MetalLB controller deployment is in running state", func() {
-				Eventually(func() bool {
+				Eventually(func() error {
 					deploy, err := testclient.Client.Deployments(metallb.Namespace).Get(context.Background(), consts.MetalLBDeploymentName, metav1.GetOptions{})
 					if err != nil {
-						return false
+						return err
 					}
-					return deploy.Status.ReadyReplicas > 0 && deploy.Status.ReadyReplicas == deploy.Status.Replicas
-				}, metallbutils.DeployTimeout, metallbutils.Interval).Should(BeTrue())
 
-				pods, err := testclient.Client.Pods(OperatorNameSpace).List(context.Background(), metav1.ListOptions{
-					LabelSelector: "component=controller"})
-				Expect(err).ToNot(HaveOccurred())
+					pods, err := testclient.Client.Pods(OperatorNameSpace).List(context.Background(), metav1.ListOptions{
+						LabelSelector: "component=controller"})
+					if err != nil {
+						return err
+					}
 
-				deploy, err := testclient.Client.Deployments(metallb.Namespace).Get(context.Background(), consts.MetalLBDeploymentName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				Expect(len(pods.Items)).To(Equal(int(deploy.Status.Replicas)))
+					if len(pods.Items) != int(deploy.Status.Replicas) {
+						return pkgerrors.Errorf("deployment %s pods are not ready, expected %d replicas got %d pods", consts.MetalLBOperatorDeploymentName, deploy.Status.Replicas, len(pods.Items))
```

Comment:
Why not `fmt.Errorf` ?

Reply:
what's the difference?

Reply:
fmt.Errorf comes from the standard library, so it's preferrable not to add a dependency that does the same thing as teh stldib